virtual-desktop/src/app/authentication-manager/login/login.component.ts

```diff
@@ -168,14 +172,33 @@ export class LoginComponent implements OnInit {
             let jsonMessage = JSON.parse(error);
             if (jsonMessage.categories) {
               let failedTypes = [];
+              let failedPlugins = new Set<string>();
               let keys = Object.keys(jsonMessage.categories);
               for (let i = 0; i < keys.length; i++) {
                 if (!jsonMessage.categories[keys[i]].success) {
                   failedTypes.push(keys[i]);
                 }
               }
-              this.errorMessage = this.translation.translate('AuthenticationFailed',
+              this.errorDetails = '';
+              let err;
+              for (let i = 0; i < failedTypes.length; i++) {
+                let plugins = Object.keys(jsonMessage.categories[failedTypes[i]].plugins);
+                for (let j = 0; j < plugins.length; j++) {
+                  err = jsonMessage.categories[failedTypes[i]].plugins[plugins[j]].error;
+                  if(err) {
+                    this.errorMessage = err.message;
```

Comment:
err.message might not exist. There is no enforcement of what is inside of `error` object.
You should set this.errorMessage to the previous default (`this.translation.translate('AuthenticationFailed', { numTypes: failedTypes.length, types: JSON.stringify(failedTypes) });`) until you find the first object which actually does have an `err.message` to replace the default.